spring20-wvd-templates/Create and provision WVD host pool/CreateHostpoolTemplate.json

```diff
@@ -0,0 +1,701 @@
+  },
+  "resources": [
+    {
+      "name": "[parameters('hostpoolName')]",
+      "apiVersion": "[parameters('apiVersion')]",
+      "type": "Microsoft.DesktopVirtualization/hostpools",
+      "location": "[parameters('location')]",
+      "tags": "[parameters('hostpoolTags')]",
+      "properties": {
+        "friendlyName": "[parameters('hostpoolFriendlyName')]",
+        "description": "[parameters('hostpoolDescription')]",
+        "hostpoolType": "[parameters('hostpoolType')]",
+        "customRdpProperty": "[parameters('customRdpProperty')]",
+        "personalDesktopAssignmentType": "[parameters('personalDesktopAssignmentType')]",
+        "maxSessionLimit": "[parameters('maxSessionLimit')]",
+        "loadBalancerType": "[parameters('loadBalancerType')]",
+        "validationEnviroment": "[parameters('validationEnviroment')]",
+        "ring": null,
+        "registrationInfo": {
+          "expirationTime": "[parameters('tokenExpirationTime')]",
+          "token": null,
+          "registrationTokenOperation": "Update"
+        },
+        "vmTemplate": "[parameters('vmTemplate')]"
+      }
+    },
+    {
+      "name": "[variables('appGroupName')]",
+      "apiVersion": "[parameters('apiVersion')]",
+      "type": "Microsoft.DesktopVirtualization/applicationgroups",
+      "location": "[parameters('location')]",
+      "tags": "[parameters('applicationGroupTags')]",
+      "properties": {
+        "hostpoolarmpath": "[resourceId('Microsoft.DesktopVirtualization/hostpools/', parameters('hostpoolName'))]",
+        "friendlyName": "Default Desktop",
+        "description": "Desktop Application Group created through the Hostpool Wizard",
+        "applicationGroupType": "Desktop"
+      },
+      "dependsOn": [
+        "[resourceId('Microsoft.DesktopVirtualization/hostpools/', parameters('hostpoolName'))]"
+      ]
+    },
+    {
+      "apiVersion": "2018-05-01",
+      "name": "[concat('Workspace-linkedTemplate-', parameters('deploymentId'))]",
+      "type": "Microsoft.Resources/deployments",
+      "resourceGroup": "[variables('workspaceResourceGroup')]",
+      "condition": "[parameters('addToWorkspace')]",
+      "properties": {
+        "mode": "Incremental",
+        "template": {
+          "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+          "contentVersion": "1.0.0.0",
+          "resources": [
+            {
+              "apiVersion": "[parameters('apiVersion')]",
+              "name": "[parameters('workSpaceName')]",
+              "type": "Microsoft.DesktopVirtualization/workspaces",
+              "location": "[parameters('workspaceLocation')]",
+              "properties": {
+                "applicationGroupReferences": "[variables('applicationGroupReferencesArr')]"
+              }
+            }

```

Comment:
Curious why is this resource and few other ones below created inside a nested template instead of the main template itself

Reply:
what do you mean this is not nested??

Reply:
@datr93 
Why is this workspace resource (Line 527) is nested under Microsoft.Resources/deployments instead of directly defining it in line 515 as part of the main template ?

Reply:
@nakranimohit0 because its needed when one is deploying in a rg thats not the main one.

Reply:
ah gotcha, thanks for clarifying it